src/goose.rs

```diff
@@ -1850,16 +1852,37 @@ pub fn get_base_url(
     config_host: Option<String>,
     task_set_host: Option<String>,
     default_host: Option<String>,
-) -> Url {
+) -> Result<Url, GooseError> {
     // If the `--host` CLI option is set, build the URL with it.
     match config_host {
-        Some(host) => Url::parse(&host).unwrap(),
+        Some(host) => Ok(
+            Url::parse(&host).map_err(|parse_error| GooseError::InvalidHost {
+                host,
+                detail: Some("Failure parsing host specified with --host".to_string()),
+                parse_error,
+            })?,
+        ),
         None => {
             match task_set_host {
                 // Otherwise, if `GooseTaskSet.host` is defined, usee this
-                Some(host) => Url::parse(&host).unwrap(),
+                Some(host) => {
+                    Ok(
+                        Url::parse(&host).map_err(|parse_error| GooseError::InvalidHost {
+                            host,
+                            detail: Some(
+                                "Failure parsing host specified with GooseTaskSet.set_host()"
+                                    .to_string(),
+                            ),
+                            parse_error,
+                        })?,
+                    )
+                }
                 // Otherwise, use global `GooseAttack.host`. `unwrap` okay as host validation was done at startup.
-                None => Url::parse(&default_host.unwrap()).unwrap(),
+                None => {
+                    // Host is required, if we get here it's safe to unwrap this variable.
+                    let default_host = default_host.unwrap();
+                    Ok(Url::parse(&default_host).map_err(|parse_error| GooseError::InvalidHost { host: default_host.to_string(), detail: Some("Failure parsing host specified globally with GooseAttack.set_host()".to_string()), parse_error })?)
```

Comment:
nit - Maybe split this line?

Reply:
Fixed. I let rustfmt handle this for me, but this line was ever-so-slightly different than the lines above so it chose to not reformat it. Doing it manually I agree it reads better.